src/nvim/plines.c

```diff
@@ -885,6 +885,9 @@ int plines_m_win(win_T *wp, linenr_T first, linenr_T last, bool limit_winheight)
     count += plines_win_full(wp, first, &next, NULL, false, limit_winheight);
     first = next + 1;
   }
+  if (first == wp->w_buffer->b_ml.ml_line_count + 1) {
+    count += win_get_fill(wp, first);
+  }
```

Comment:
Over at vim/vim it was [suggested](https://github.com/vim/vim/pull/14410#discussion_r1554805999) to not add this to `plines_win()`, and decided to instead add it to `plines_m_win()`. Going by [this](https://github.com/vim/vim/pull/14410#discussion_r1554853125) logic, it might make more sense for us to add it to `plines_win_full()`? Or do we still have concerns of it causing problems there?

Reply:
But that'll lead to incorrect `w_cline_height`.